sources/Algo/include/Algo.h

```diff
@@ -393,5 +398,300 @@ class ControllerInterfaceDefinitionAlgorithm : public NodeAlgorithm {
+   */
+  static boost::optional<boost::filesystem::path> computeLibPath(const std::string& lib);
+
+ private:
+  /// @brief Extract models from configuration before processing the nodes
+  void extractDynModels();
+
+  /**
+   * @brief Process single association from configuratio
+   *
+   * @param automaton the config automaton definition
+   * @param macro the macro connection connecting to the singleassociation
+   * @param singleassoc the single association config element to process
+   */
+  void dispatchAutomatonSingle(const inputs::AssemblingXmlDocument::DynamicAutomaton& automaton, const inputs::AssemblingXmlDocument::MacroConnect& macro,
+                               const inputs::AssemblingXmlDocument::SingleAssociation& singleassoc);
+
+  /**
+   * @brief Process multi association from configuratio
+   *
+   * @param automaton the config automaton definition
+   * @param macro the macro connection connecting to the singleassociation
+   * @param multiassoc the multi association config element to process
+   */
+  void dispatchAutomatonMulti(const inputs::AssemblingXmlDocument::DynamicAutomaton& automaton, const inputs::AssemblingXmlDocument::MacroConnect& macro,
+                              const inputs::AssemblingXmlDocument::MultipleAssociation& multiassoc);
+
+  /**
+   * @brief Process node in case of automaton bus connection
+   * @param node node to process
+   */
+  void processDynModelBusConnection(const NodePtr& node);
+
+  /**
+   * @brief Process node in case of automaton shunt connection
+   * @param node node to process
+   */
+  void processDynModelShuntConnection(const NodePtr& node);
+
+  /**
+   * @brief Process node in case of automaton line connection
+   * @param line line to process
+   */
+  void processDynModelLineConnection(const std::shared_ptr<inputs::Line>& line);
+
+  /**
+   * @brief Process node in case of automaton transfo connection
+   * @param tfo transfo to process
+   */
+  void processDynModelTfoConnection(const std::shared_ptr<inputs::Tfo>& tfo);
+
+  /**
+   * @brief Add macro connection to node connections of the automaton definition
+   *
+   * Creates the dynamic model definition if not already existing
+   * Update the dynamic model definitions member
+   *
+   * @param automaton the config automaton
+   * @param macroConnection the macro connection to add
+   */
+  void addMacroConnectionToDef(const dfl::inputs::AssemblingXmlDocument::DynamicAutomaton& automaton,
+                               const DynModelDefinition::MacroConnection& macroConnection);
+
+ private:
+  DynModelDefinitions& models_;  ///< automaton definitions to update
+
+  std::unordered_map<std::string, inputs::AssemblingXmlDocument::DynamicAutomaton> automatonsById_;  ///< config automaton by model id
+  std::unordered_map<inputs::VoltageLevel::VoltageLevelId, std::unordered_set<MacroConnect, MacroConnectHash>>
+      macroConnectByVlForBusesId_;  ///< macro connections for buses, by voltage level
```

Comment:
Cannot we find something more elegant than have 4 hashmaps?

Reply:
Each of the maps contain different information so I don't think we can avoid that